bgpd/bgp_nht.c

```diff
@@ -813,3 +822,30 @@ static void path_nh_map(struct bgp_info *path, struct bgp_nexthop_cache *bnc,
 		path->nexthop->path_count++;
 	}
 }
+
+/*
+ * This function is called to register nexthops to zebra
+ * as that we may have tried to install the nexthops
+ * before we actually have a zebra connection
+ */
+void bgp_nht_register_nexthops(struct bgp *bgp)
```

Comment:
This is more for my understanding. Shouldn't we need similar unregister function ?

Reply:
The normal shutdown takes care of this, this is taking care of bgp being created and attempting to talk to zebra before zebra is ready to go.